server/chat.ts

```diff
@@ -1647,6 +1652,18 @@ export const Chat = new class {
 			entry.hasRoomPermissions = /\bthis\.can\([^,)\n]*, [^,)\n]*,/.test(handlerCode);
 			entry.broadcastable = /\bthis\.(?:canBroadcast|runBroadcast)\(/.test(handlerCode);
 
+			// assign SOME properties from the base command if the current command uses CommandContext.run.
+			const runsCommand = /this.run\((?:'|"|`)(.*?)(?:'|"|`)\)/.exec(handlerCode);
+			if (runsCommand) {
+				const [, , baseCommand] = runsCommand;
+				const baseEntry = commandTable[baseCommand];
+				if (baseEntry) {
+					entry.requiresRoom = baseEntry.requiresRoom;
+					entry.hasRoompermissions = baseEntry.hasRoomPermissions;
```

Comment:
```suggestion
					entry.hasRoomPermissions = baseEntry.hasRoomPermissions;
```

`entry` having type `any` means you need to be extra careful here.